slackbot/src/kagent_slackbot/slack/formatters.py

```diff
@@ -0,0 +1,379 @@
+"""Slack Block Kit formatting"""
+
+import re
+from typing import Optional, Any
+from datetime import datetime
+from ..constants import SLACK_BLOCK_LIMIT, EMOJI_ROBOT, EMOJI_CLOCK
+
+
+def needs_approval(response_text: str) -> bool:
+    """
+    Detect if agent response needs human approval
+
+    Args:
+        response_text: Agent's response text
+
+    Returns:
+        True if approval is needed
+    """
+    approval_patterns = [
```

Comment:
this isn't a flexible way to do the approval checks -- I'd rather remove it and once we figure out a model for proper elicitations, then we should implement it. I am afraid that doing regex for certain phrases might not work too well and have some false positives perhaps.